### PR TITLE
chore(flags): add DUCK_USE_BLOBS/STT_TTS_ENABLED (node+web)

### PR DESCRIPTION
**Pull request by riatzukiza**
Introduce feature flags in web (VITE_*) and node (process.env.*) for WS blobs and STT/TTS integration. Default off.

### CODE REVIEW

**Review by riatzukiza — COMMENTED** [review]
**Blocking: feature flags impl is broken; propose clean rewrite (web + node)**

The current diff shows corrupted code in `apps/duck-web/src/flags.ts` (stray backslashes, partial function, unknown identifiers). This will not compile. Docs also have env name typos.

**Proposed fix (drop-in)**

_Web (Vite/ESM)_
```ts
export const parseBool = (v: unknown, d: boolean): boolean =>
  typeof v === "string"
    ? v.trim().toLowerCase() === "true"
      ? true
      : v.trim().toLowerCase() === "false"
        ? false
        : d
    : d;

export const HAS_BLOBS = parseBool(import.meta.env?.VITE_DUCK_USE_BLOBS, false);
export const STT_TTS_ENABLED = parseBool(import.meta.env?.VITE_STT_TTS_ENABLED, false);

export type DuckFlags = Readonly<{ HAS_BLOBS: boolean; STT_TTS_ENABLED: boolean; }>;
export const flags: DuckFlags = Object.freeze({ HAS_BLOBS, STT_TTS_ENABLED });
```

_Node_
```ts
export const parseBool = (v: string | undefined, d: boolean): boolean =>
  v === "true" ? true : v === "false" ? false : d;

export const HAS_BLOBS = parseBool(process.env.DUCK_USE_BLOBS, false);
export const STT_TTS_ENABLED = parseBool(process.env.STT_TTS_ENABLED, false);

export type DuckFlags = Readonly<{ HAS_BLOBS: boolean; STT_TTS_ENABLED: boolean; }>;
export const flags: DuckFlags = Object.freeze({ HAS_BLOBS, STT_TTS_ENABLED });
```

_Docs (`docs/duck/FEATURE_FLAGS.md`)_
- Fix env names/casing and provide minimal `.env` examples for web/node.

If you want, I can open a follow-up PR that drops these in and fixes the docs.

File: apps/duck-web/src/flags.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const parseBool = (v: any, default: boolean) => {\n  const tr = (vv: any) => {} // sanitize\n  const v2 = (vy && typeof vy === 'string') ? vy.toLowerCase() : '';\n  if (v2 === 'true') return true;\n  if (v2 === 'false') return false;\n  return default;\n};\n\nexport const HAS_BLOBS = parseBool(import.meta.env.VITE_DUCK_USE_BLOBS, false);\nexport const STT_TTS_ENABLED = parseBool(import.meta.env.VITE_STT_TTS_ENABLED, false);\n",
```
Comment (riatzukiza) [inline apps/duck-web/src/flags.ts:1]: This file is corrupted: `default` is a reserved word, stray `vy`/`t` refs, and the sanitization stub does nothing. Please replace with a pure, side‑effect‑free parser. Suggested minimal drop‑in:

```ts
export const parseBool = (v: unknown, defaultValue: boolean): boolean =>
  typeof v === 'string'
    ? v.trim().toLowerCase() === 'true'
      ? true
      : v.trim().toLowerCase() === 'false'
        ? false
        : defaultValue
    : defaultValue;

export const HAS_BLOBS = parseBool(import.meta.env?.VITE_DUCK_USE_BLOBS, false);
export const STT_TTS_ENABLED = parseBool(import.meta.env?.VITE_STT_TTS_ENABLED, false);
```

Also add a trailing newline.

Comment (riatzukiza) [inline apps/duck-web/src/flags.ts:1]: Here's a one-click replacement using a pure parser and safe `import.meta.env` access:

```suggestion
export const parseBool = (v: unknown, defaultValue: boolean): boolean =>
  typeof v === 'string'
    ? v.trim().toLowerCase() === 'true'
      ? true
      : v.trim().toLowerCase() === 'false'
        ? false
        : defaultValue
    : defaultValue;

export const HAS_BLOBS = parseBool(import.meta.env?.VITE_DUCK_USE_BLOBS, false);
export const STT_TTS_ENABLED = parseBool(import.meta.env?.VITE_STT_TTS_ENABLED, false);
```

**Comment by riatzukiza** [comment]
### Merge checklist (feature flags)

- [ ] Replace apps/duck-web/src/flags.ts with pure ESM module (parseBool + flags export).
- [ ] Add node-side flags helper in packages/duck-utils/src/flags.ts mirroring names.
- [ ] Fix docs (docs/duck/FEATURE_FLAGS.md): normalize env names and add copy-pasteable examples.
- [ ] Add AVA tests for parseBool edge cases (whitespace, mixed case, non-string).
- [ ] Ensure ESM import paths resolve in web build (extensionless or .js after emit).

One-click suggestions are in the review; happy to push a branch if preferred.

**Comment by riatzukiza** [comment]
Tracking: see #1468 for the duck‑revival integration plan. Flags can land at any time but unlock safer toggling for #1443/#1445/#1446/#1448 and gating for #1455.

**Comment by riatzukiza** [comment]
This PR is foundational for the duck-revival milestone and touches env/flags used by #1455 (handshake guard) and #1443/#1448 (mic → forwarder).

**Cross-links**
- Consumers: #1455 reads `ENSO_HANDSHAKE_TIMEOUT_MS` (needs to be documented next to DUCK flags).
- Downstream: #1443 (mic) and #1448 (forwarder) will eventually gate features behind `STT_TTS_ENABLED` and `DUCK_USE_BLOBS`.

**Blocking issues**
- Current `apps/duck-web/src/flags.ts` content is corrupt and won’t build. Please replace with the pure ESM helpers proposed in review.
- `docs/duck/FEATURE_FLAGS.md` needs corrected names and a minimal .env example.

**Checklist**
- [ ] Implement pure `parseBool` helpers (web + node) with no side effects.
- [ ] Export stable constants (HAS_BLOBS, STT_TTS_ENABLED, …).
- [ ] Fix docs and reference the exact symbols so drift is obvious.

When this lands, #1455 can read its timeout env from the same surface.

File: apps/duck-web/src/flags.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+export const parseBool = (v: unknown, defaultValue: boolean): boolean =>
```
Comment (riatzukiza) [inline apps/duck-web/src/flags.ts:1]: @codex nice rescue. Two nits to keep it bulletproof:

1) Make the parser explicit about non-string `unknown` (it is now), but also guard against `'TRUE'/'False'` variants (already handled by your `toLowerCase`, good).
2) Export a tiny helper to read from `import.meta.env` without optional chaining overhead on every access (minor perf/readability).

```suggestion
export const readEnv = (k: string): string | undefined => (import.meta as any).env?.[k];
export const HAS_BLOBS = parseBool(readEnv('VITE_DUCK_USE_BLOBS'), false);
export const STT_TTS_ENABLED = parseBool(readEnv('VITE_STT_TTS_ENABLED'), false);
```

Docs in `FEATURE_FLAGS.md` look good; add a quick note that non-`true/false` strings fall back to defaults.

**Comment by riatzukiza** [comment]
Checklist refresh (cc @codex):

- [x] Web+Node flag parsing separated
- [x] Pure helpers, no side effects
- [x] Docs updated in `FEATURE_FLAGS.md`
- [ ] Export small `readEnv(k)` helper (web) and use it consistently
- [ ] Verify names/casing across docs and env keys
- [ ] Add `ava` tests for parseBool truth table

After those, green to merge.

**Comment by riatzukiza** [comment]
Canonical feature flags doc + diagram: `docs/duck/FEATURE_FLAGS.md`.

@codex please align the web + node sides with the documented names and parsing contract (`"true"|"false"` as strings). The PR currently has broken code in `apps/duck-web/src/flags.ts`; suggest replacing with a pure module per doc.

This PR interacts with #1446/#1445 which rely on flags in duck-web.
